### ISSUE TITLE
Deserializing boolean values

### ISSUE THREAD

**Issue by anto-acube** [goetas-webservices/xsd2php#120]
In the UBL 2.1 Invoice I got this XSD definition

```
<xsd:element name="ChargeIndicator" type="ChargeIndicatorType"/>
```

Which maps to a complex type `udt:IndicatorType`

```
<xsd:complexType name="ChargeIndicatorType">
      <xsd:simpleContent>
         <xsd:extension base="udt:IndicatorType"/>
      </xsd:simpleContent>
   </xsd:complexType>
```

Which maps to

```
<xsd:complexType name="IndicatorType">
    <xsd:annotation>
      <xsd:documentation xml:lang="en">
        <ccts:UniqueID>UBLUDT0000012</ccts:UniqueID>
        <ccts:CategoryCode>UDT</ccts:CategoryCode>
        <ccts:DictionaryEntryName>Indicator. Type</ccts:DictionaryEntryName>
        <ccts:VersionID>1.0</ccts:VersionID>
        <ccts:Definition>A list of two mutually exclusive Boolean values that express the only possible states of a property.</ccts:Definition>
        <ccts:RepresentationTermName>Indicator</ccts:RepresentationTermName>
        <ccts:PrimitiveType>string</ccts:PrimitiveType>
      </xsd:documentation>
    </xsd:annotation>
    <xsd:simpleContent>
      <xsd:extension base="xsd:boolean"/>
    </xsd:simpleContent>
  </xsd:complexType>
```

The generated PHP class using the `ChargeIndicator` has

```
   class AllowanceChargeType {

   /**
     * @var bool $chargeIndicator
     */
    private $chargeIndicator = null;
    ...

    /**
     * Gets as chargeIndicator
     *
     * @return bool
     */
    public function getChargeIndicator()
    {
        return $this->chargeIndicator;
    }

    /**
     * Sets a new chargeIndicator
     *
     * @param bool $chargeIndicator
     * @return self
     */
    public function setChargeIndicator($chargeIndicator)
    {
        $this->chargeIndicator = $chargeIndicator;
        return $this;
    }

    ...
```

Now if i deserialize an XML which contains

```
...
<cac:AllowanceCharge>
    <cbc:ChargeIndicator>false</cbc:ChargeIndicator>
    <cbc:AllowanceChargeReason>Promotion discount</cbc:AllowanceChargeReason>
    <cbc:Amount currencyID="EUR">100</cbc:Amount>
</cac:AllowanceCharge>
...
```

and inspect the generated object, I got that `$object->getChargeIndicator()` is of type `ChargeIndicator` and its `__value` property is `NULL`, no matter the value the XML contains.

Any hint why this is happening?

Thanks in advance

**Comment by goetas** [comment]
Can you please share also the generated YAML metadata?

**Comment by anto-acube** [comment]
Hi! Here is the `AllowanceCharge` metadata

```
Model\CommonAggregate\AllowanceChargeType:
    properties:
        ...
        chargeIndicator:
            expose: true
            access_type: public_method
            serialized_name: ChargeIndicator
            xml_element:
                namespace: 'urn:oasis:names:specification:ubl:schema:xsd:CommonBasicComponents-2'
            accessor:
                getter: getChargeIndicator
                setter: setChargeIndicator
            type: Model\CommonBasic\ChargeIndicator
        ...
```

The `ChargeIndicator` definition

```
Model\CommonBasic\ChargeIndicator:
    xml_root_name: ChargeIndicator
    xml_root_namespace: 'urn:oasis:names:specification:ubl:schema:xsd:CommonBasicComponents-2'
```

And the `ChargeIndicatorType` definition

```
Model\CommonBasic\ChargeIndicatorType:
    properties: {  }
```

**Comment by goetas** [comment]
Hmm, that seems a bug. `Model\CommonBasic\ChargeIndicatorType` should not be generated at all, and in `Model\CommonAggregate\AllowanceChargeType`, the propv 'chargeIndicator` should be just `type: boolean`

**Comment by anto-acube** [comment]
Thanks @goetas ! I did the changes as you suggested. Thanks for your help

**Comment by goetas** [comment]
The yaml files should be auto updated. Re opening as this is probably a bug

**Comment by goetas** [comment]
Fixed in 0.4.0
